nucypher/cli/commands/stake.py

```diff
@@ -1308,24 +1309,43 @@ def events(general_config, staker_options, config_file, event_name):
         individual_allocation=STAKEHOLDER.individual_allocation,
         force=True)
 
-    title = f" {STAKEHOLDER.staking_agent.contract_name} Events ".center(40, "-")
-    emitter.echo(f"\n{title}\n", bold=True, color='green')
     if event_name:
-        events = [STAKEHOLDER.staking_agent.contract.events[event_name]]
+        event_type = STAKEHOLDER.staking_agent.contract.events[event_name]
     else:
         raise click.BadOptionUsage(message="You must specify an event name with --event-name")
         # TODO: Doesn't work for the moment
         # event_names = STAKEHOLDER.staking_agent.events.names
         # events = [STAKEHOLDER.staking_agent.contract.events[e] for e in event_names]
         # events = [e for e in events if 'staker' in e.argument_names]
 
-    for event in events:
-        emitter.echo(f"{event.event_name}:", bold=True, color='yellow')
-        entries = event.getLogs(fromBlock=0, toBlock='latest', argument_filters={'staker': staking_address})
+    argument_filters = {'staker': staking_address}
+    contract_name = STAKEHOLDER.staking_agent.contract_name
+    if csv_file:
+        _write_events_csv_file(csv_file, argument_filters, event_type)
+        emitter.echo(f"\n{contract_name}::{event_name} events written to {csv_file}", bold=True, color='green')
+    else:
+        title = f" {contract_name} Events ".center(40, "-")
+        emitter.echo(f"\n{title}\n", bold=True, color='green')
+        emitter.echo(f"{event_type.event_name}:", bold=True, color='yellow')
+        entries = event_type.getLogs(fromBlock=0, toBlock='latest', argument_filters=argument_filters)
         for event_record in entries:
             emitter.echo(f"  - {EventRecord(event_record)}")
 
 
+def _write_events_csv_file(csv_file, event_args_filter, event_type):
```

Comment:
Consider relocating this method to a helper module.

Reply:
✔️